src/main/java/tools/jackson/core/util/BufferRecyclerPool.java

```diff
@@ -0,0 +1,13 @@
+package tools.jackson.core.util;
+
+public class BufferRecyclerPool {
+
+    private static final ObjectPool<BufferRecycler> pool = ObjectPool.newLockFreePool(BufferRecycler::new);
```

Comment:
Ok; since this is now global pool (across all factories), there definitely should be maximum size. But then again, having such setting would lead to need for configuration.
And static singletons are generally a bad idea and not suitable for use by (low-level) libraries.

In fact, now that I think about this, I think pooling (or default implementation at any rate) really needs to be segmented by factory instance -- no buffers should be shared (by default impl) across `TokenStreamFactory` instances.